### PR TITLE
Yacarof/tdp 4696 refactor export step

### PR DESCRIPTION
**Pull request by yacarof**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4696

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1161/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1161/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1162/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1162/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1163/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1163/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1164/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1164/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1165/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1165/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1167/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1167/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1168/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1168/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1169/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1169/)

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java [hunk 1]
```diff
@@ -21,35 +22,16 @@
...
-                params.get(CSV_FIELDS_DELIMITER), filename, params.get(CSV_ESCAPE_CHARACTER_PARAM),
-                params.get(CSV_ENCLOSURE_CHARACTER_PARAM), params.get(CSV_ENCLOSURE_MODE_PARAM), params.get(CSV_CHARSET_PARAM))
-                .asInputStream();
+        ExportType exportType = epAnalyzer.detectExportType(params);
 
-        // store the body content in a temporary File
-        File tempFile = api.storeInputStreamAsTempFile(filename, csv);
-        context.storeTempFile(filename, tempFile);
+        ExportSampleStep exporter = epAnalyzer.getExporter(exportType);
+        exporter.exportSample(params);
```
Comment (fabballe) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java]: getExporter haas a @Nullable so you have to test if `exporter` is not null before use it

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java [hunk 2]
```diff
@@ -21,35 +22,16 @@
     /** This class' logger. */
     private static final Logger LOGGER = LoggerFactory.getLogger(ExportPreparationStep.class);
```
Comment (fabballe) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java]: LOGGER is never used

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportParamAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    private ExportSampleStepCSV exportSampleStepCSV;
+
+    @Nullable
+    public ExportType detectExportType(@NotNull Map<String, String> params) {
+        String exportType = params.get(MandatoryParameters.EXPORT_TYPE.getName());
+        return ExportType.getExportType(exportType);
+    }
+
+    @NotNull
+    public EnumMap<MandatoryParameters, String> extractMandatoryParameters(@NotNull Map<String, String> params) {
```
Comment (fabballe) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportParamAnalyzer.java]: `extractMandatoryParameters` seems to be never call

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportSampleParamCSV.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    /**
+     * Get a corresponding {@link ExportSampleParamCSV} from a {@link String}.
+     *
+     * @param pName the {@link ExportSampleParamCSV#name}.
+     * @return the corresponding {@link ExportSampleParamCSV} or <code>null</code> if there isn't.
+     */
+    @Nullable
+    public static ExportSampleParamCSV getExportSampleParamCSV(@NotNull String pName) {
```
Comment (fabballe) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportSampleParamCSV.java]: Same for `getExportSampleParamCSV` method

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java (hunk 2 above)
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java]: Done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java (hunk 1 above)
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/ExportPreparationStep.java]: Done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportParamAnalyzer.java (hunk 3 above)
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportParamAnalyzer.java]: Done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportSampleParamCSV.java (hunk 4 above)
Comment (yacarof) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/util/export/ExportSampleParamCSV.java]: Done.

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1198/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1198/)

**Pull request merged**